app/lib/features/chat_ng/utils.dart

```diff
@@ -2,6 +2,11 @@ import 'package:acter/common/providers/chat_providers.dart';
 import 'package:acter/features/chat_ng/providers/chat_room_messages_provider.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 
+const userMentionMarker = '‖';
+final userMentionRegExp = RegExp(
+  r'\[@([^\]]+)\]\(https:\/\/matrix\.to\/#\/(@[^)]+)\)',
+);
```

Comment:
there is also `matrix:`-urls as valid user mentions...

and I thought we were not doing regexps anymore but had an actual DOM model we'd be doing stuff on?

In particular I would expect most of these features to move _into_ the mentions-plugin / the editor side. As from the UI perspective it is just a link that is going to be rendered differently due to the editor doing something different...

Reply:
Not exactly sure what that DOM model is, can you share reference or source? Here it's only being used because of the message format is in markdown and we could detect mention patterns. I am happy to change it based upon improved and safe version

Reply:
I think this is work for a future PR ...